tests/test_cos_sin_mean_wave_direction.py

```diff
@@ -0,0 +1,115 @@
+
+SIN_MWD_VALUES = np.array([[0.44817262, 0.52311930], [0.96465370, 0.98718704], [0.90645754, 0.79016611]])
+
+
+@pytest.fixture
+def mwd_source():
+    MWD_SPEC = [
+        {"param": "mwd", "values": MWD_VALUES, **MOCK_FIELD_METADATA},
+    ]
+    return source_registry.create("testing", fields=MWD_SPEC)
+
+
+@pytest.fixture
+def cos_sin_mwd_source():
+    COS_SIN_MWD = [
+        {"param": "cos_mwd", "values": COS_MWD_VALUES, **MOCK_FIELD_METADATA},
+        {"param": "sin_mwd", "values": SIN_MWD_VALUES, **MOCK_FIELD_METADATA},
+    ]
+    return source_registry.create("testing", fields=COS_SIN_MWD)
+
+
+def test_cos_sin_mean_wave_direction(mwd_source):
+    """Test the cos_sin_mean_wave_direction filter."""
+    filter = filter_registry.create(
+        "cos_sin_mean_wave_direction",
+        mean_wave_direction="mwd",
+        cos_mean_wave_direction="cos_mwd",
+        sin_mean_wave_direction="sin_mwd",
+    )
+    pipeline = mwd_source | filter
+
+    output_fields = collect_fields_by_param(pipeline)
+
+    assert set(output_fields) == {"cos_mwd", "sin_mwd"}
+    assert len(output_fields["cos_mwd"]) == 1
+    assert len(output_fields["sin_mwd"]) == 1
+
+    assert np.allclose(output_fields["cos_mwd"][0].to_numpy(), COS_MWD_VALUES)
+    assert np.allclose(output_fields["sin_mwd"][0].to_numpy(), SIN_MWD_VALUES)
+
+
+def test_cos_sin_mean_wave_direction_reverse(cos_sin_mwd_source):
+    """Test the cos_sin_mean_wave_direction filter in reverse."""
+    filter = filter_registry.create(
+        "cos_sin_mean_wave_direction",
+        mean_wave_direction="mwd",
+        cos_mean_wave_direction="cos_mwd",
+        sin_mean_wave_direction="sin_mwd",
+    ).reverse()
+    pipeline = cos_sin_mwd_source | filter
+
+    output_fields = collect_fields_by_param(pipeline)
+
+    assert set(output_fields) == {"mwd"}
+    assert len(output_fields["mwd"]) == 1
+
+    assert np.allclose(output_fields["mwd"][0].to_numpy(), MWD_VALUES)
+
+
+def test_cos_sin_mean_wave_direction_round_trip(mwd_source):
+    """Test the cos_sin_mean_wave_direction filter reproduces inputs on a round trip."""
+    filter = filter_registry.create(
+        "cos_sin_mean_wave_direction",
+        mean_wave_direction="mwd",
+        cos_mean_wave_direction="cos_mwd",
+        sin_mean_wave_direction="sin_mwd",
+    )
+    cos_sin_mwd_source = mwd_source | filter
+    pipeline = cos_sin_mwd_source | filter.reverse()
+
+    input_fields = collect_fields_by_param(mwd_source)
+    intermediate_fields = collect_fields_by_param(cos_sin_mwd_source)
+    output_fields = collect_fields_by_param(pipeline)
+
+    assert set(input_fields) == {"mwd"}
+    assert set(intermediate_fields) == {"cos_mwd", "sin_mwd"}
+    assert set(output_fields) == {"mwd"}
+
+    for input_field, output_field in zip(input_fields["mwd"], output_fields["mwd"]):
+        assert_fields_equal(input_field, output_field)
+
+
+if __name__ == "__main__":
+    """
+    Run all test functions that start with 'test_'.
```

Comment:
this is just a curiosity question - but if the tests are run with `pytest` do we need this?